frontend/public/components/impersonate-notifier.jsx

```diff
@@ -1,24 +1,33 @@
-import * as _ from 'lodash-es';
 import * as React from 'react';
 import { connect } from 'react-redux';
+import { useTranslation, Trans } from 'react-i18next';
 
 import * as UIActions from '../actions/ui';
+import { modelFor } from '../module/k8s';
 
 export const ImpersonateNotifier = connect(({ UI }) => ({ impersonate: UI.get('impersonate') }), {
   stopImpersonate: UIActions.stopImpersonate,
 })(({ stopImpersonate, impersonate }) => {
+  const { t } = useTranslation();
   if (!impersonate) {
     return null;
   }
   return (
     <div className="co-global-notification">
       <div className="co-global-notification__content">
         <p className="co-global-notification__text">
-          <span className="text-uppercase co-global-notification__impersonate-kind">{`Impersonating ${impersonate.kind}`}</span>{' '}
-          You are impersonating{' '}
-          <span className="co-global-notification__impersonate-name">{impersonate.name}</span>. You
-          are viewing all resources and roles this {_.toLower(impersonate.kind)} can access.{' '}
-          <a onClick={stopImpersonate}>Stop Impersonation</a>
+          <span className="text-uppercase co-global-notification__impersonate-kind">
+            {t('public~Impersonating {{kind}}', {
+              kind: t(modelFor(impersonate.kind)?.labelKey) || impersonate.kind,
```

Comment:
Have we tested what happens if we pass `undefined` to `t`? I suspect it will trigger a key not found error. It would be better to get the model earlier and have logic like

```suggestion
              kind: model?.labelKey ? t(model.labelKey) : impersonate.kind,
```

Reply:
@spadgett yes, I've tested that, and passing undefined will return an empty string which will result in the right hand side of the || being assigned ie impersonate.kind. But I can change up the syntax if you think its clearer.

Reply:
assigning this to a const since we're doing it twice.